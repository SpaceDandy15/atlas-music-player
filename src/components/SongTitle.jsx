import React from "react";

export default function SongTitle({ title = "Song Title", artist = "Artist" }) {
  return (
    <div className="flex flex-col items-start text-left">
      <h2 className="text-lg font-bold text-white -ml-1">{title}</h2>
      <p className="text-sm text-[var(--color-tahiti)] -ml-1">{artist}</p>
    </div>
  );
}
