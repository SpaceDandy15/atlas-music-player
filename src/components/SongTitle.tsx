import React from "react";

type SongTitleProps = {
  title?: string;
  artist?: string;
};

export default function SongTitle({
  title = "Song Title",
  artist = "Artist",
}: SongTitleProps) {
  return (
    <div className="flex flex-col items-start text-left">
      <h2 className="text-lg font-bold text-white -ml-1">{title}</h2>
      <p className="text-sm text-[var(--color-tahiti)] -ml-1">{artist}</p>
    </div>
  );
}
