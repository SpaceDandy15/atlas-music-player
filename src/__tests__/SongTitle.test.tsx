import React from "react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import SongTitle from "../components/SongTitle";

describe("SongTitle Component", () => {
  it("renders correctly with default props", () => {
    // Snapshot test to ensure the component renders correctly with default title and artist
    const tree = renderer.create(<SongTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom title and artist", () => {
    // Snapshot test to ensure that passing props updates the rendered output
    const tree = renderer
      .create(<SongTitle title="Custom Song" artist="Custom Artist" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
