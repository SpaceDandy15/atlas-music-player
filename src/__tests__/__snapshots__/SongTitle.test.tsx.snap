// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > renders correctly with custom title and artist 1`] = `null`;

exports[`SongTitle Component > renders correctly with default props 1`] = `null`;
